mesmer/stats/_linear_regression.py

```diff
@@ -88,30 +98,45 @@ def predict(
         if available_predictors - required_predictors:
             superfluous = sorted(available_predictors - required_predictors)
             superfluous = "', '".join(superfluous)
-            raise ValueError(f"Superfluous predictors: '{superfluous}'")
+            raise ValueError(
+                f"Superfluous predictors: '{superfluous}', either params",
+                "for this predictor are missing or you forgot to add it to 'exclude'.",
+            )
 
         if "intercept" in exclude:
             prediction = xr.zeros_like(params.intercept)
         else:
             prediction = params.intercept
 
+        # if predictors is a DataTree, rename all data variables to "pred" to avoid conflicts
+        # not necessaey if predictors is empty DataTree or only data is in root, i.e. depth == 0
+        if isinstance(predictors, DataTree) and not predictors.depth == 0:
+            predictors = map_over_subtree(
+                lambda ds: ds.rename({var: "pred" for var in ds.data_vars})
+            )(predictors)
```

Comment:
```suggestion
            (var,) = ds.data_vars
            predictors = map_over_subtree(
                lambda ds: ds.rename({var: "pred"})
            )(predictors)
```
to make it more obvious that it's only one variable?

Reply:
ds is not defined outside of map_over_subtree sadly, one could do `ds.rename({list(ds.data_vars)[0]: "pred"})`

Reply:
Ah yes, or go with a function instead of a lambda. I'll leave it up to you